appium-tests/helpers/cameraHelper.js

```diff
@@ -103,11 +109,185 @@ module.exports.getPicture = function (opts, pid) {
     }, opts);
 };
 
-module.exports.checkPicture = function (pid, cb) {
+// verifies taken picture when the promise is resolved,
+// calls a callback with 'OK' if everything is good,
+// calls a callback with 'ERROR: <error message>' if something is wrong
+// note that this function is executed in the context of tested app
+// and not in the context of tests
+module.exports.checkPicture = function (pid, options, cb) {
+    var isIos = cordova.platformId === "ios";
+    var isAndroid = cordova.platformId === "android";
+    // skip image type check if it's unmodified on Android:
+    // https://github.com/apache/cordova-plugin-camera/#android-quirks-1
+    var skipFileTypeCheck = isAndroid &&
+        (!options.quality || options.quality === 100) &&
```

Comment:
Should the default be 50? I believe the intent was that it will use lesser memory and hence, default of 50 is a good thing - but I did some profiling and setting the quality to anything but 100 on Android actually causes more memory consumption because the image has to be loaded as a bitmap and re-compressed. Setting appropriate target height & width along with FILE_URI is the correct way to reduce the memory requirements. I propose that the default be changed to 100.